grammar.y

```diff
@@ -393,35 +383,86 @@ table_alias:
 ;
 
 join_clause:
-  table_expr JOIN table_expr join_constraint
```

Comment:
This is the core of this PR. We changed the token `JOIN` for a grammar rule `join_op`